### DISCUSSION TITLE
Questions about Atmos Home.

### DISCUSSION THREAD

**Discussion by joaoserra** [VoidXH/Cavern#145] [General]
Hi, so i've read this(https://github.com/VoidXH/Cavern/blob/master/docs/Guidelines%20for%20object-based%20movie%20audio.md) and your last post on this thread (https://www.audiosciencereview.com/forum/index.php?threads/atmos-finally-decoded-in-pc-mac.32351/page-16), specifically this part:´

_"So, back to the topic, we have this 9.1.6 maximum. Why shouldn't they just pre-render that? Well, they do! Your mix is pre-made, sounds exactly the same on that layout, saves computing power, allows for transport over ARC, etc. Every lower channel count available for home users can be losslessly matrix mixed from 9.1.6, the formulas yield the exact same result after these 2 transformations compared to just mixing straight to a lower channel count."_

I want to make sure that i understood what you said correctly: Basically you are saying that if you render any Atmos Home EAC3/JOC mix (limited to 16 audio essences) to 9.1.6 and then downmix it, using the formulas you mentioned to, say 5.1.4 regular channels.  This downmixed 5.1.4 will give you the exact same result that a native render of the same content 5.1.4 would, right?

If that's the case, i have a few questions:

- Besides marketing(or visualizers) , why would Dolby add this kind of complexity to Home Atmos? What not simply use discrete 9.1.6 that could be downmixed losslessly to any lower channel count? Do you have any idea? 
- Where are those downmixing matrixes documented? 

Thanks!

**Comment by VoidXH** [comment]
> This downmixed 5.1.4 will give you the exact same result that a native render of the same content 5.1.4 would, right?

Yes. Downmixing transformations where no new channels appear, just old ones disappear, are converging to the same result, regardless of which ones you remove in each step. Even if you do 9.1.6 to 9.1.4, then 7.1.4, then 5.1.4, you'll get the same results as simply mixing to 5.1.4. If you remove a channel pair, they will be spread between neighbours, which the mixer does.

> What not simply use discrete 9.1.6 that could be downmixed losslessly to any lower channel count? Do you have any idea?

It has some optimization possibilities, sometimes solving the full soundspace from less objects, and thus, less bandwidth. It also has the upside of sometimes giving better directivity with more speakers. Its effect and market worth is questionable, but is there scientifically nonetheless. Empty objects can just be turned off, no other codec supported this before, and just stored a lot of zeros uselessly. There is a very simple reason too, people just wanted it. While objects are needed for large rooms like commercial cinemas, not the home, but people demanded the same. They got it, even if it doesn't make a difference.

> Where are those downmixing matrixes documented?

While those aren't documented, the exact method is. I only know it's balance based, just like Cavern, but the mixing curves are unknown. I use constant power, and while it doesn't match up with DRP, it sounds more precise. If you downmix channels with the same curve, it yields the exact same result as mixing there. Look at Cavern/Source.cs for the exact algorithm.

**Comment by joaoserra** [comment]
Thanks for the anwsers!

> It has some optimization possibilities, sometimes solving the full soundspace from less objects, and thus, less bandwidth. It also has the upside of sometimes giving better directivity with more speakers. Its effect and market worth is questionable, but is there scientifically nonetheless. Empty objects can just be turned off, no other codec supported this before, and just stored a lot of zeros uselessly. There is a very simple reason too, people just wanted it. While objects are needed for large rooms like commercial cinemas, not the home, but people demanded the same. They got it, even if it doesn't make a difference.

What about room correction algorithms? Would having access to the room correction data help the renderer? Or is room correction in AVRs only applied after the rendering?  
Based on what you have seen, have you actually seen a movie with 9.1.6 beds? Or most of them typically use 7.1.2 plus 6 objects?

**Comment by VoidXH** [comment]
> Or is room correction in AVRs only applied after the rendering?

Room correction is for already rendered channels since they correct for a single speaker.

> Based on what you have seen, have you actually seen a movie with 9.1.6 beds? Or most of them typically use 7.1.2 plus 6 objects?

There is a YouTube channel called "Object Demo", you can see examples for both there. The former is common for old movies and demo discs (Star Wars VII or Ready Player One), and 7.1 + 8 objects is getting common for new movies. The Adam Project is one example of very heavy object use, that's the example animation on Cavern's website.

**Comment by joaoserra** [comment]
Thanks for the clarification.

Just one more thing: What's your thoughts on IAMF? https://aomediacodec.github.io/iamf/

This is backed by Samsung/Google and if you look closely on the github repo, will find also spec contributions by Netflix and Apple.
They are also adding support for it for FFmpeg (https://www.phoronix.com/news/FFmpeg-Immersive-IAMF) and according to this PR (https://aomedia.org/press%20releases/AOMedia-Advances-the-Audio-Innovation-Era/) they are also supporting ADM based workflows/content creation. via EAR Production Suite (https://ear-production-suite.ebu.io/)
I guess IAMF will be a thing!

They are supporting channel and scene(ambiosonics) based audio (but not objects, likely to avoid Dolby/DTS patents)
I'm curious on your thoughts on this.

**Comment by VoidXH** [comment]
> What's your thoughts on IAMF?

![this](https://imgs.xkcd.com/comics/standards_2x.png)

> They are also adding support for it for FFmpeg

Isn't that channel-based support only? FFmpeg has no object pipeline to my knowledge.

> I guess IAMF will be a thing!

For ambisonics on PC, yes, it's very likely. But there is no planned AVR support at all, so Atmos will stay the only option for speaker layouts unless soundcard makers start to add more Jack outputs.

**Comment by joaoserra** [comment]
Their main goal is to avoid Dolby licensing. This is being developed via AOMedia, same as AV1

> Isn't that channel-based support only? FFmpeg has no object pipeline to my knowledge.
> 
> >
Channel and Ambiosonics. IAMF doesn't support objects (likely to avoid Dolby/DTS patents)



> For ambisonics on PC, yes, it's very likely. But there is no planned AVR support at all, so Atmos will stay the only option for speaker layouts unless soundcard makers start to add more Jack outputs.
> 
> >
I don't think so. AndroidTV is a target. About AVR support, they could pre-render all the stuff and output via 5.1.2/5.1.4/9.1.4 LPCM (Not sure if AVRs support this)
But more likely, they will probably require HW manufacturers (At least STBs and TVs, and AVRs will come) to support it, like they do with mandatory AV1 support. 
If you are a HW manufacture and want to have YouTube and Netflix apps, they will only certify new hardware with AV1 support. From what i heard, Google will also force HDR10+ on Youtube certification.  (They refuse to support Dolby Vision)
Google/Samsung/Apple/Netflix can really force this if they really want, and they really want to bypass Dolby licensing...
